modules/assim.sequential/R/sda.enkf_MultiSite.R

```
@@ -317,22 +317,40 @@ sda.enkf.multisite <- function(settings,
   # weight matrix
   wt.mat <- matrix(NA, nrow = nens, ncol = nt)
   # Reading param samples------------------------------- 
-  #create params object using samples generated from TRAITS functions
+  # create params object using samples generated from TRAITS functions
   if(restart_flag){
     new.params <- new.params
   } else {
-    if(!file.exists(file.path(settings$outdir, "samples.Rdata"))) PEcAn.logger::logger.severe("samples.Rdata cannot be found. Make sure you generate samples by running the get.parameter.samples function before running SDA.")
-    #Generate parameter needs to be run before this to generate the samples. This is hopefully done in the main workflow.
-    if(is.null(ensemble.samples)){
+    generate_samples <- FALSE
+    if (is.null(ensemble.samples)) {
+      if (file.exists(file.path(settings$outdir, "samples.Rdata"))) {
+        load(file.path(settings$outdir, "samples.Rdata"))
+      } else {
+        generate_samples <- TRUE
+      }
+    }
+    # get the joint input design.
+    # here we are looping over sites
+    # to make sure we are grabbing the complete input lists.
+    for (i in seq_along(settings)) {
```

I'm not following the logic here. First, why is this in a loop over sites? Second, you've added a lot of checks to make sure that there's a complete mapping between the input space and the samplingspace, but you're not returning any information to the user if that's not the case. Related, I'm not following the expectation that if the samplingspace and inputs don't match for the first site that we should just keep looping over sites -- why would the input space change by site, and if it did you'd still have the problem that the input_design would work for some sites but not others. 

I'll also note that it's not obvious where the input_design is being saved

---

1. The `for-loop` over sites is because under some edge cases where some sites may not have the entire input lists, which are required in the `generate_joint_ensemble_design` function. For example, we should have `parameters`, `ICs`, `met`, and `soil_physics` as inputs, but some sites may not have `soil_physics`. Therefore, we will not have the complete inputs' index table if we use any of those sites to run the `generate_joint_ensemble_design ` function.
2. I don't think we need to return any information for sites that don't have any of the above inputs, except for the `met` inputs, for which the user will see the error in the SIPNET logfile. It's natural for some sites that are missing any of the inputs, cause the datasets that are used for creating those files are not available everywhere on Earth.
3. The SDA will fail only if any site doesn't have the met files. Otherwise, the workflow should just work as expected. Even for some sites that don't have some inputs (`soil_physics` or `ICs`), the workflow should just pick the default value of SIPNET and not fail.

---

I don't agree. If a site doesn't have multiple soil_physics files, but then the input_design tells ensemble member i to run with soil_physics file k, then the write.configs is going to crash when it tries to access a non-existent file path. I'm also not following why we'd be doing an ensemble run where some sites were missing inputs, that seems like something that should throw an error not something we should create a workaround for. The behavior of silently reverting to the default SIPNET values is a dangerous one -- those defaults are not reliable general-purpose values but the parameter values specific to Niwot Ridge, CO. And just because you've created that workaround in SIPNET doesn't  mean the same workaround exists in the write.configs for other models.

---

Ok. I just reverted all the changes to the SDA workflow. The error will be reported in the `write.config` function if there is any mismatch between the samplingspace and input lists.

---

I didn't mean that the checks should be dropped -- I actually thought that the checks were a useful feature. It seems better to me to catch a mismatch here during the setup than after spawning 100 x 8000 runs.

---

Fixed. I included code that checks for any mismatch between the sampling space and the site's input list. If there is any mismatch, the user will know which site is missing which input from the `mis.match.table`.